src/Microsoft.DncEng.SecretManager/ConsoleExtension.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.DncEng.CommandLineLib;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.DncEng.SecretManager
+{
+    internal static class ConsoleExtension
+    {
+        const int DefaultRetries = 3;
+
+        public static Task<string> PromptAndValidateAsync(this IConsole console, string fieldName, string help, Func<string, bool> validation)
+        {
+            return PromptAndValidateAsync(console, fieldName, help, validation, l => l);
+        }
+
+        public static async Task<T> PromptAndValidateAsync<T>(this IConsole console, string fieldName, string help, Func<string, bool> validate, Func<string, T> parse)
```

Comment:
This might be easier with a delegate with out parameter here so you don't have to pass both a TryParse call, and the equivalend Parse call. Instead you can just make one lambda with TryParse in it.

Reply:
Good idea, I added delegate TryParse so we don't need to do double calls.